packages/better-auth/src/plugins/oidc-provider/index.ts

```diff
@@ -1301,14 +1301,31 @@ export const oidcProvider = (options: OIDCOptions) => {
 					const session = await getSessionFromCtx(ctx);
 
 					// Check authorization
-					if (!session && !options.allowDynamicClientRegistration) {
```

Comment:
This line of code allows the creation of public clients using `dynamicClientRegistration`. The structure is intensional even though the `dynamic` might not be the best variable name here.

Reply:
So users should be able to create a client even when unauthenticated?